fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -203,6 +214,65 @@ public boolean createTable(CreateTableStmt stmt) throws DdlException {
         return icebergCatalog.createTable(dbName, tableName, schema, partitionSpec, tableLocation, createTableProperties);
     }
 
+    @Override
+    public void alterTable(AlterTableStmt stmt) throws UserException {
+        String dbName = stmt.getDbName();
+        String tableName = stmt.getTableName();
+        org.apache.iceberg.Table table = icebergCatalog.getTable(dbName, tableName);
+
+        if (table == null) {
+            throw new StarRocksConnectorException(
+                    "Failed to load iceberg table: " + stmt.getTbl().toString());
+        }
+
+        List<AlterClause> alterClauses = stmt.getOps();
+        List<AlterClause> tableChanges = Lists.newArrayList();
+        List<AlterClause> schemaChanges = Lists.newArrayList();
+        for (AlterClause clause : alterClauses) {
+            if (clause instanceof AddColumnClause
+                    || clause instanceof AddColumnsClause
+                    || clause instanceof DropColumnClause
+                    || clause instanceof ColumnRenameClause
+                    || clause instanceof ModifyColumnClause) {
+                schemaChanges.add(clause);
+            } else if (clause instanceof ModifyTablePropertiesClause
+                    || clause instanceof TableRenameClause
+                    || clause instanceof AlterTableCommentClause
+            ) {
+                tableChanges.add(clause);
+            } else {
+                throw new StarRocksConnectorException(
+                        "Unsupported alter operation for iceberg connector: " + clause.toString());
+            }
+        }
+
+        commitAlterTable(table, schemaChanges, tableChanges);
+
+        synchronized(this) {
+            tables.remove(TableIdentifier.of(dbName, tableName));
+            icebergCatalog.refreshTable(dbName, tableName, jobPlanningExecutor);
```

Comment:
catch refresh table exception and then invalidate the cache of cachingIcebergCatalog.